std/zip.d

```diff
@@ -823,6 +825,78 @@ final class ZipArchive
     {
         data[i .. i + 8] = nativeToLittleEndian(ul);
     }
+
+    /* ============== for detecting overlaps =============== */
+
+private:
+
+    // defines a segment of the zip file, including start, excluding end
+    struct Segment
+    {
+        uint start;
+        uint end;
+    }
+
+    // removes Segment start .. end from _segs
+    // throws zipException if start .. end is not completely available in _segs;
+    void removeSegment(uint start, uint end) pure @safe
+    {
+        if (_segs.length == 0)
+            throw new ZipException("overlapping data detected");
+
+        uint best = 0;
+        size_t bestPos = 0;
+        foreach (i,seg;_segs)
+            if (seg.start <= start && seg.start > best)
```

Comment:
`seg.start > best` will fail to find a segment that starts at offset 0 but is not the first segment in `_segs`. Not sure if this will affect how you will use this.

Reply:
This is indeed a bug. I'll fix this in a few hours.